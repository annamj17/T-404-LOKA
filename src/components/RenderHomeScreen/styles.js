import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  screens: {
    flexDirection: 'column',
    height: '100%',
    width: '100%'
  },
  iconsContainer: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignSelf: 'center',
    // paddingTop: 30,
    width: 270,
    height: '100%'
  },
  icon: {
    fontSize: 90
  },
  image: {
    marginTop: 30,
    height: '45%',
    width: '90%',
    alignSelf: 'center',
    borderRadius: 10
  },
  textCss: {
    fontSize: 25,
    alignSelf: 'center',
    marginTop: 200,
    color: '#318F95',
    fontFamily: 'Helvetica',
    fontWeight: 'bold'
  },
  imageContainer: {
    marginTop: 'auto',
    height: '12%'
  },
  tagItem: {
    alignSelf: 'center',
    paddingTop: 20,
    width: 250,
    height: 50,
    marginBottom: 10,
    fontSize: 18,
    color: 'grey',
    fontFamily: 'Helvetica',
    fontWeight: 'bold'
  },
  confirmContainer: {
    alignItems: 'center',
    justifyContent: 'center',
    alignSelf: 'center',
    backgroundColor: '#318F95',
    borderRadius: 10,
    marginTop: 30,
    height: 60,
    width: 200
  },
  confirmText: {
    color: 'white',
    fontFamily: 'Helvetica',
    fontWeight: 'bold',
    fontSize: 20
  }
});
