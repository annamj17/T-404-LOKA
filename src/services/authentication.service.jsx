import axios from 'axios';
import { AsyncStorage } from 'react-native';
import FormData from 'form-data';

import { API_URL } from '../constants/index';

const TokenTypes = {
  PortalAccessToken: 'portalAccessToken',
  PortalRefreshToken: 'portalRefreshToken',
  LanguageCode: 'languageCode',
  TokenType: 'tokenType',
  TenantId: 'tenantId'
};

export const getPortalAccessToken = () => AsyncStorage.getItem(TokenTypes.PortalAccessToken);

export const getPortalRefreshToken = () => AsyncStorage.getItem(TokenTypes.PortalRefreshToken);

export const getTokenType = () => AsyncStorage.getItem(TokenTypes.TokenType);

const setTokenData = data => {
  AsyncStorage.setItem(TokenTypes.PortalAccessToken, data.access_token.toString());
  AsyncStorage.setItem(TokenTypes.PortalRefreshToken, data.refresh_token.toString());
  AsyncStorage.setItem(TokenTypes.LanguageCode, data.languageCode.toString());
  AsyncStorage.setItem(TokenTypes.TokenType, data.token_type.toString());
  AsyncStorage.setItem(TokenTypes.TenantId, data.tenantId.toString());
};

export const logout = () => {
  AsyncStorage.removeItem(TokenTypes.PortalAccessToken);
  AsyncStorage.removeItem(TokenTypes.PortalRefreshToken);
};

export const getNewAccessToken = async refreshToken => {
  let response;
  const formData = new FormData();
  formData.append('refresh_token', refreshToken);
  formData.append('grant_type', 'refresh_token');

  try {
    response = await axios.post(`${API_URL}/token`, formData);
  } catch (error) {
    logout();
  }

  const { status, data } = response;

  // If refresh token is invalid. The server will return with status 200, invalid_grant
  if (status === 200 && data.error === 'invalid_grant') {
    logout();
  } else {
    setTokenData(data);
  }

  return data;
};

export const logIn = async (username, password) => {
  let response;

  const formData = new FormData();
  formData.append('username', username);
  formData.append('password', password);
  formData.append('grant_type', 'password');

  try {
    response = await axios.post(`${API_URL}/token`, formData);
  } catch (error) {
    logout();
    return false;
  }

  const { status, data } = response;

  if (status && status === 200 && !data.error) {
    setTokenData(data);
    return true;
  }

  return false;
};
