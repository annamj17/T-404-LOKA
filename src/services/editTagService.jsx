import api from '../utils/api';

// Sends edited tags to the gateway and recieves edited tags back
export const getAllTagsAfterEdit = async (editedProductTags, id) => {
  let response;

  try {
    response = await api.put(`/image/${id}`, editedProductTags);
  } catch (error) {
    return false;
  }
  const editedTags = response.data;
  return editedTags;
};

export default getAllTagsAfterEdit;
