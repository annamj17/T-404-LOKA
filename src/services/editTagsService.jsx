const ENDPOINT = '';

const getAllCinemas = async () => {
  // eslint-disable-next-line no-undef
  return fetch(ENDPOINT, {
    method: 'GET',
    headers: {}
  })
    .then(response => response.json())
    .then(responseData => {
      return responseData;
    });
};

export default getAllCinemas;
