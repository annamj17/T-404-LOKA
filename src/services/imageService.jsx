import * as ImagePicker from 'expo-image-picker';
import * as Permissions from 'expo-permissions';

// Prompt the user for types of permissions
const getPermission = async permissionTypes => {
  await Promise.all(
    permissionTypes.map(async type => Permissions.askAsync(type))
  );
};

// Selecting a photo from the cameraroll
// Uses getPermission to ask for the permission type for reading or writing to the camera roll.
export const selectFromCameraRoll = async () => {
  await getPermission([Permissions.CAMERA_ROLL]);
  const result = await ImagePicker.launchImageLibraryAsync({
    mediaTypes: ImagePicker.MediaTypeOptions.Images,
    quality: 0.8,
    base64: true,
    aspect: [16, 9]
  });

  if (result.cancelled) {
    return '';
  }
  return result.uri;
};

// Takes a photo using your camera
// Uses getPermission to ask for the permission type for photo and video taking
export const takePhoto = async () => {
  await getPermission([Permissions.CAMERA, Permissions.CAMERA_ROLL]);
  const result = await ImagePicker.launchCameraAsync({
    mediaTypes: ImagePicker.MediaTypeOptions.Images,
    quality: 0.8,
    base64: true,
    aspect: [16, 9]
  });

  if (result.cancelled) {
    return '';
  }
  return result.uri;
};
