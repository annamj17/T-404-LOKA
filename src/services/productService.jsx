/* eslint-disable no-undef */

// GATEWAY ENDAPUNKTUR
const ENDPOINT = '';

const getAllTagsForProduct = async () => {
  return fetch(ENDPOINT, {
    method: 'GET',
    headers: {}
  })
    .then(response => response.json())
    .then(responseData => {
      return responseData;
    });
};

export default { getAllTagsForProduct };
