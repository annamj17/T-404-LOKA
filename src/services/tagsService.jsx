import api from '../utils/api';

// Sends an image to the gateway and recieves all tags as response
export const getAllTagsForProduct = async image => {
  let response;

  try {
    response = await api.post(`/image`, { image });
  } catch (error) {
    return false;
  }
  const tags = response.data;
  return tags;
};

export default getAllTagsForProduct;
