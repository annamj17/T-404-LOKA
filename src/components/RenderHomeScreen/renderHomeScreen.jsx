import React from 'react';
import { View, Text, TouchableOpacity, Image } from 'react-native';
import { Entypo } from '@expo/vector-icons';

import { takePhoto, selectFromCameraRoll } from '../../services/imageService';
import styles from './styles';
import bottomImage from '../../../assets/Capture.png';

class RenderHomeScreen extends React.Component {
  // Calls takePhoto from imageService
  async takePhoto() {
    const { navigation } = this.props;
    const photo = await takePhoto();
    if (photo !== null) {
      const thePhoto = photo.photoToDisplay;
      const theTags = photo.tags;
      navigation.navigate('ProductView', { thePhoto, theTags });
    } else {
      navigation.navigate('HomeView');
    }
  }

  // Calls selectFromCameraRoll from imageService
  async selectFromCameraRoll() {
    const { navigation } = this.props;
    const photo = await selectFromCameraRoll();
    if (photo !== null) {
      const thePhoto = photo.photoToDisplay;
      const theTags = photo.tags;
      navigation.navigate('ProductView', { thePhoto, theTags });
    } else {
      navigation.navigate('HomeView');
    }
  }

  render() {
    return (
      <View style={styles.screens}>
        <View>
          <Text style={styles.textCss}>Add New Item</Text>
          <View style={styles.iconsContainer}>
            <TouchableOpacity name="camera" onPress={() => this.takePhoto()}>
              <Entypo style={styles.icon} name="camera" />
            </TouchableOpacity>
            <TouchableOpacity name="cameraRoll" onPress={() => this.selectFromCameraRoll()}>
              <Entypo style={styles.icon} name="image" />
            </TouchableOpacity>
          </View>
        </View>
        <Image style={styles.imageContainer} source={bottomImage} resizeMode="stretch" />
      </View>
    );
  }
}

export default RenderHomeScreen;
