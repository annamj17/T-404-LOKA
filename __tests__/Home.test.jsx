import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import RenderHomeScreen from '../src/components/RenderHomeScreen/renderHomeScreen';

import { selectFromCameraRoll, takePhoto } from '../src/services/imageService';

jest.mock('../src/services/imageService');

describe('Should render the Home page without crashing', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(<RenderHomeScreen />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});

describe('HomeScreen tests', () => {
  const onNavigateSpy = jest.fn();
  const props = {
    navigation: {
      navigate: onNavigateSpy
    }
  };
  const photo = {
    photoToDisplay: 'mynd.jpg',
    tags: {
      productDetailsId: '1',
      color: 'black',
      colorScore: '99.98',
      brand: 'Nike',
      brandScore: '99.78',
      type: 'Sweatshirt',
      typeScore: '98.99',
      productPrice: { fullConditionPrice: '50', resalePrice: '25' }
    }
  };

  const wrapper = shallow(<RenderHomeScreen {...props} {...photo} />);

  it('Should find 2 TouchableOpacity', () => {
    expect(wrapper.find('TouchableOpacity')).toHaveLength(2);
  });

  it('Should handle selectFromCameraRoll onPress and navigation', async () => {
    await selectFromCameraRoll.mockReturnValueOnce({
      photo: {
        photoToDisplay: 'mynd.jpg',
        tags: {
          productDetailsId: '1',
          color: 'black',
          colorScore: '99.98',
          brand: 'Nike',
          brandScore: '99.78',
          type: 'Sweatshirt',
          typeScore: '98.99',
          productPrice: { fullConditionPrice: '50', resalePrice: '25' }
        }
      }
    });

    const wrapp = shallow(<RenderHomeScreen {...props} {...photo} />);
    wrapp.find('TouchableOpacity[name="cameraRoll"]').simulate('Press');
    expect(photo).not.toBeNull();
  });

  it('Should handle takePhoto onPress and navigation', async () => {
    await takePhoto.mockReturnValueOnce({
      photo: {
        photoToDisplay: 'mynd.jpg',
        tags: {
          productDetailsId: '1',
          color: 'black',
          colorScore: '99.98',
          brand: 'Nike',
          brandScore: '99.78',
          type: 'Sweatshirt',
          typeScore: '98.99',
          productPrice: { fullConditionPrice: '50', resalePrice: '25' }
        }
      }
    });

    const wrapp = shallow(<RenderHomeScreen {...props} {...photo} />);
    wrapp.find('TouchableOpacity[name="camera"]').simulate('Press');
    expect(photo).not.toBeNull();
  });
});
