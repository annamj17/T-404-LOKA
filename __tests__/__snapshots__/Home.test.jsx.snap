// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the Home page without crashing renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#318F95",
          "fontSize": 25,
          "fontWeight": "bold",
          "marginTop": 200,
        }
      }
    >
      Add New Item
    </Text>
    <View
      style={
        Object {
          "alignSelf": "center",
          "flexDirection": "row",
          "height": "100%",
          "justifyContent": "space-around",
          "width": 270,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
  <Image
    resizeMode="stretch"
    source={1}
    style={
      Object {
        "height": "12%",
        "marginTop": "auto",
      }
    }
  />
</View>
`;
